restli-server/src/main/java/com/linkedin/restli/internal/server/model/RestLiAnnotationReader.java

```diff
@@ -2694,9 +2697,226 @@ else if (CollectionResult.class.isAssignableFrom(returnType))
+   * throw a {@link ResourceConfigException}.</p>
+   */
+  private static void validateLinkedBatchFinder(ResourceMethodDescriptor finder,
+      ResourceModel resourceModel, String linkedBatchFinderName)
+  {
+    ResourceMethodDescriptor batchFinder =
+        resourceModel.findBatchFinderMethod(linkedBatchFinderName);
+
+    if (batchFinder == null)
+    {
+      throw new ResourceConfigException("Did not find any Linked @BatchFinder method named '"
+          + linkedBatchFinderName
+          + "' from @Finder Method named '" + finder.getFinderName()
+          + "' on class '"
+          + resourceModel.getResourceClass().getName() + '\'');
+    }
+
+    ParameterizedType collectionType = (ParameterizedType) getLogicalReturnType(batchFinder.getMethod());
+    final Class<?> batchFinderCriteriaType = (Class<?>) collectionType.getActualTypeArguments()[0];
+    final Class<?> batchFinderMetadataType = (Class<?>) collectionType.getActualTypeArguments()[2];
+    final Class<?> finderMetadataType = finder.getCollectionCustomMetadataType();
+    if (finderMetadataType != null && !finderMetadataType.equals(batchFinderMetadataType))
+    {
+      throw new ResourceConfigException("Linked @BatchFinder method '" + linkedBatchFinderName
+          + "' from @Finder Method named '" + finder.getFinderName()
+          + "' on class '" + resourceModel.getResourceClass().getName()
+          + "' does not have the same metadata type as the finder.");
+    }
+
+    final RecordDataSchema criteriaSchema = (RecordDataSchema) DataTemplateUtil.getSchema(batchFinderCriteriaType);
+
+    // Check that there is only parameter defined on the batch finder.
+    if (batchFinder.getParameters().size() != 1)
+    {
+      throw new ResourceConfigException("Linked @BatchFinder method '" + linkedBatchFinderName
+          + "' from @Finder Method named '" + finder.getFinderName()
+          + "' on class '" + resourceModel.getResourceClass().getName()
+          + "' has more than 1 parameter.");
+    }
+
+    // Check that the schema of the parameter is an array of the criteria schema.
+    DataSchema batchFinderParamSchema = batchFinder.getParameters().get(0).getDataSchema();
+    if (batchFinderParamSchema.getType() != DataSchema.Type.ARRAY
+        || !((ArrayDataSchema) batchFinderParamSchema).getItems().equals(criteriaSchema))
+    {
+      throw new ResourceConfigException("Linked @BatchFinder method '" + linkedBatchFinderName
+          + "' from @Finder Method named '" + finder.getFinderName()
+          + "' on class '" + resourceModel.getResourceClass().getName()
+          + "' parameter is not an array of the criteria schema");
+    }
+
+    Set<String> fieldNames = criteriaSchema.getFields()
+        .stream()
+        .map(RecordDataSchema.Field::getName)
+        .collect(Collectors.toCollection(HashSet::new));
+    finder.getParameters().forEach(parameter ->
+    {
+      switch (parameter.getParamType())
+      {
+        case QUERY:
+        case KEY:
```

Comment:
I think that should be ASSOC_KEY, not KEY

Reply:
KEY (@AssocKey) is deprecated version of ASSOC_KEY (@AssocKeyParam)